File: src/features/typing/data/java/maximum_subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int best = nums[0];
        int sum = 0;
        for (int num : nums) {
            sum = Math.max(sum, 0) + num;
            best = Math.max(best, sum);
        }
        return best;
    }
}

File: src/features/typing/data/dart/maximum_subarray.dart
class Solution {
  int maxSubArray(List<int> nums) {
    var best = nums[0];
    var sum = 0;
    for (final num in nums) {
      if (sum < 0) {
        sum = 0;
      }
      sum += num;
      if (sum > best) {
        best = sum;
      }
    }
    return best;
  }
}
